Description: Ignore zero-length devices
 There seems no point in complaining about these.  The problem with throwing
 an exception is that it's difficult for partitioners such as partman to
 programmatically distinguish between this kind of thing and some more
 serious error that actually merits letting the user know.
Bug-Debian: http://bugs.debian.org/602533
Forwarded: http://lists.gnu.org/archive/html/bug-parted/2010-11/msg00011.html
Author: Colin Watson <cjwatson@ubuntu.com>
Reviewed-by: Otavio Salvador <otavio@ossystems.com.br>
Last-Update: 2010-11-19

Index: b/libparted/arch/linux.c
===================================================================
--- a/libparted/arch/linux.c
+++ b/libparted/arch/linux.c
@@ -1180,6 +1180,12 @@
         if (_device_probe_geometry (dev)) {
                 ped_exception_leave_all ();
         } else {
+                if (!_device_get_length (dev)) {
+                        ped_exception_catch ();
+                        ped_exception_leave_all ();
+                        goto error_close_dev;
+                }
+
                 /* hack to allow use of files, for testing */
                 ped_exception_catch ();
                 ped_exception_leave_all ();
